### PR TITLE
Matrix implementation of dct with unit test

### PR DESCRIPTION
**Pull request by adanlopeza**

### CODE REVIEW

File: xla/jfdctxla.c [hunk 1]
```diff
@@ -7,21 +7,85 @@
...
+	11.096320, 15.391046, 14.498035, 13.047892, 11.096315, 8.718322,  6.005284,  3.061468,
+	10.452501, 14.498041, 13.656854, 12.290866, 10.452503, 8.212481,  5.656853,  2.883838,
+	9.407003,  13.047894, 12.290842, 11.061483, 9.407002,  7.391037,  5.091034,  2.595389,
+	8.000001,  11.096319, 10.452504, 9.407007,  8.000000,  6.285560,  4.329569,  2.207195,
+	6.285562,  8.718323,  8.212480,  7.391035,  6.285561,  4.938532,  3.401721,  1.734182,
+	4.329566,  6.005280,  5.656854,  5.091043,  4.329567,  3.401721,  2.343146,  1.194525,
+	2.207195,  3.061468,  2.883839,  2.595387,  2.207199,  1.734182,  1.194525,  0.608964
+};
+
+FAST_FLOAT tmp[DCTSIZE * DCTSIZE];
```
Comment (itaiz-google) [inline xla/jfdctxla.c]: Are you sure this code isn't going to be used in a multi-threaded context?

File: xla/jfdctxla.c [hunk 2]
```diff
@@ -7,21 +7,85 @@
...
+	memcpy(result, tmp, sizeof tmp);
+}
 
 GLOBAL(void)
 jpeg_fdct_xla(FAST_FLOAT *data)
 {
-
+	fprintf(stderr, "In xla dct\n");
+	matrix_multiply(dct_matrix, data, data);
+	matrix_multiply(data, transposed_dct_matrix, data);
```
Comment (itaiz-google) [inline xla/jfdctxla.c]: If you do this:

matrix_multiply(dct_matrix, data, tmp);
matrix_multiply(tmp, transposed_dct_matrix, data);

then you don't need the matrix copy on line 77.

File: xla/jfdctxla.c [hunk 3]
```diff
@@ -7,21 +7,85 @@
...
+	
+	int i;
```
Comment (itaiz-google) [inline xla/jfdctxla.c:78]: Is this redundant?

File: xla/dctunittest.c [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		for(j = 0; j < DCTSIZE; j++) {
+			data[j] = (FAST_FLOAT) rand() / RAND_MAX * 200;
+		}
+}
+
+int main(int argc, char *argv[])
+{
+	FAST_FLOAT block[DCTSIZE * DCTSIZE];
+	int i;
+	srand((unsigned)time(NULL));
```
Comment (itaiz-google) [inline xla/dctunittest.c]: Might want to control the seed for predictability.

**Review by itaiz-google — APPROVED** [review]

File: xla/dctunittest.c [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+/*
+ * Create an 8x8 matrix with random values.
+ */
+static void fill_block(FAST_FLOAT *data) {
+	int i, j;
+	for(i = 0; i < DCTSIZE; i++, data += DCTSIZE)
+		for(j = 0; j < DCTSIZE; j++) {
+			data[j] = (FAST_FLOAT) rand() / RAND_MAX * 200;
```
Comment (ghost) [inline xla/dctunittest.c]: What is the origin of the 200?

Comment (ghost) [reply xla/dctunittest.c]: Correct to -128 to 127.

**Review by ghost — APPROVED** [review]
LGTM.

**Pull request merged**